azure/converters/subnets.go

```diff
@@ -32,3 +33,16 @@ func GetSubnetAddresses(subnet network.Subnet) []string {
 	}
 	return addresses
 }
+
+// GetSubnetAddressesV2 returns the address prefixes contained in an SDK v2 subnet.
+func GetSubnetAddressesV2(subnet armnetwork.Subnet) []string {
```

Comment:
why the function duplication here?

Reply:
Because other go packages in CAPZ still call the SDK v1 version of `GetSubnetAddresses`, and I don't want to increase the scope of this PR to include other services. It's a temporary pattern: once all services are updated, we can get rid of the v1 version of this function (and some others that follow a similar pattern).